monai/transforms/utils.py

```diff
@@ -211,6 +211,51 @@ def map_binary_to_indices(
     return fg_indices, bg_indices
 
 
+def weighted_patch_samples(
+    spatial_size: Union[int, Sequence[int]],
+    w: np.ndarray,
+    n_samples: int = 1,
+    r_state: Optional[np.random.RandomState] = None,
+) -> List:
+    """
+    Computes `n_samples` of random patch sampling locations, given the sampling weight map `w` and patch `spatial_size`.
+
+    Args:
+        spatial_size: length of each spatial dimension of the patch.
+        w: weight map, the weights must be non-negative. each element denotes a sampling weight of the spatial location.
+            0 indicates no sampling.
+            The weight map shape is assumed ``(spatial_dim_0, spatial_dim_1, ..., spatial_dim_n)``.
+        n_samples: number of patch samples
+        r_state: a random state container
+
+    Returns:
+        a list of `n_samples` N-D integers representing the spatial sampling location of patches.
+
+    """
+    if w is None:
+        raise ValueError("w must be an ND array.")
+    if r_state is None:
+        r_state = np.random.RandomState()
+    img_size = np.asarray(w.shape, dtype=np.int)
+    win_size = np.asarray(fall_back_tuple(spatial_size, img_size), dtype=np.int)
+
+    s = tuple(slice(w // 2, m - w + w // 2) if m > w else slice(m // 2, m // 2 + 1) for w, m in zip(win_size, img_size))
```

Comment:
Won't `slice(m // 2, m // 2 + 1)` always be a 1-sized slice?

Reply:
yes, when the user-specified patch size is larger than the original image I think this central slice is a good sampling location

Reply:
Perhaps `resize_center` could be used to simplify this code here.

Reply:
thanks I wasn't aware of the utility. here the slicing is just a view change, `resize_center `  makes a new copy which has larger footprint. if you agree I can revise the util to have a no copy option and use it

Reply:
No copy might be useful elsewhere so yes go for it, or a separate function if the code gets messy.

Reply:
I couldn't use it directly in the transform, but still try to make a in place option for resize_center, could you review?